File: solutions/go/clock/4/clock.go
package clock

import "fmt"

type Clock struct {
    hours, minutes int
}

func getTime(v int) (hours, minutes int) {
    v %= 24 * 60
    v += 24 * 60
    v %= 24 * 60
    hours, minutes = v/60, v%60
    return
}

func New(hours, minutes int) Clock {
    hours, minutes = getTime(60 * hours + minutes)
	return Clock{
        hours,
        minutes,
    }
}

func (c Clock) Add(minutes int) Clock {
    c.hours, c.minutes = getTime(minutes + c.minutes + 60 * c.hours)
    return c
}

func (c Clock) Subtract(minutes int) Clock {
	return c.Add(-minutes)
}

func (c Clock) String() string {
	return fmt.Sprintf("%02d:%02d", c.hours, c.minutes)
}

File: solutions/kotlin/clock/2/src/main/kotlin/Clock.kt
import java.util.Objects

class Clock(var hours: Int, var minutes: Int) {
    val realTimeInMinute: Int
        get() = (hours * 60 + minutes).mod(24 * 60)

    val realHours: Int
        get() = realTimeInMinute / 60

    val realMinutes: Int
        get() = realTimeInMinute % 60
    
    fun subtract(minutes: Int) {
        this.minutes -= minutes
    }

    fun add(minutes: Int) {
        this.minutes += minutes
    }

    override fun toString(): String = "${realHours.toString().padStart(2, '0')}:${realMinutes.toString().padStart(2, '0')}"

    override fun equals(other: Any?): Boolean = other is Clock && other.realHours == realHours && other.realMinutes == realMinutes

    override fun hashCode(): Int = Objects.hash(realHours, realMinutes)
}